XamlControlsGallery/ControlPages/ListViewPage.xaml.cs

```diff
@@ -56,6 +83,242 @@ private void SelectionModeComboBox_SelectionChanged(object sender, SelectionChan
+                    }
+
+                    // Don't go out of bounds
+                    index = Math.Min(DragDropListView2.Items.Count, index);
+
+                    // Finally, add to target listview
+                    contacts2.Insert(index, temp);
+
+                    // Go through source list and remove the items that are being moved
+                    foreach (Contact contact in DragDropListView.Items)
+                    {
+                        if (contact.FirstName == temp.FirstName && contact.LastName == temp.LastName && contact.Company == temp.Company)
+                        {
+                            contacts1.Remove(contact);
+                            break;
+                        }
+                    }
+                }
+                
+                e.AcceptedOperation = DataPackageOperation.Move;
+                def.Complete();
+            }
+        }
+
+        private async void Source_Drop(object sender, DragEventArgs e)
+        {
+
+            var def = e.GetDeferral();
+            var s = await e.DataView.GetTextAsync();
+            var items = s.Split('\n');
+            foreach (var item in items)
+            {
+                var info = item.Split(" ", 3);
+
+                // Create Contact object from string, add to existing target ListView
+                Contact temp = new Contact(info[0], info[1], info[2]);
+
+                // Find the insertion index
+                Windows.Foundation.Point pos = e.GetPosition(DragDropListView.ItemsPanelRoot);
+
+                // Find height of a random item from the target list view
+                ListViewItem sampleItem = (ListViewItem)DragDropListView.ContainerFromIndex(0);
+                double itemHeight = sampleItem.ActualHeight + sampleItem.Margin.Top + sampleItem.Margin.Bottom;
+
+                // Find index based on dividing number of items by height of each item
+                int index = Math.Min(DragDropListView.Items.Count - 1, (int)(pos.Y / itemHeight));
+
+                // Find the item that we want to drop
+                ListViewItem targetItem = (ListViewItem)DragDropListView.ContainerFromIndex(index); ;
+
+                // Figure out if to insert above or below
+                Windows.Foundation.Point positionInItem = e.GetPosition(targetItem);
+                if (positionInItem.Y > itemHeight / 2)
+                {
+                    index++;
+                }
+
+                // Don't go out of bounds
+                index = Math.Min(DragDropListView.Items.Count, index);
+
+                // Finally, add to target listview
+                contacts1.Insert(index, temp);
+
+                // Go through source list and remove the items that are being moved
+                foreach (Contact contact in DragDropListView2.Items)
+                {
+                    if (contact.FirstName == temp.FirstName && contact.LastName == temp.LastName && contact.Company == temp.Company)
```

Comment:
Do you really need **all** these checks? Does the data provide any sort of unique ID that you can use instead?

Reply:
I went over it a bunch of times but to me it seems like this is the only way to get the desired position and match that to an index for insertion to the list. I tested with and without the last few checks as well (after determining index), but those are necessary to insert to the bottom of a list.